app/javascript/packages/prompt-on-navigate/README.md

```diff
@@ -0,0 +1,28 @@
+# `@18f/identity-prompt-on-navigate`
+
+Configures an `onbeforeunload` event handler such that the browser will prompt the user before they reload or navigate away from the page.
+
+## Usage
+
+```js
+import { promptOnNavigate } from "@18f/identity-prompt-on-navigate";
+
+// Set the onbeforeunload event handler.
+const cleanUp = promptOnNavigate();
+
+// ...some time later, call cleanUp() to restore any previous onbeforeunload handler and cancel any pending timers (this is important).
+cleanUp();
+```
+
+## Analytics
+
+By default, `promptOnNavigate` will call `trackEvent` to log a `User prompted before navigation` event when the onbeforeunload handler is called. It will then log `User prompted before navigation and still on page` events at 5, 15, and 30 seconds after the onbeforeunload handler is called (the `seconds` property on the event will contain the number of seconds since the initial prompt).
```

Comment:
Just to confirm my understanding of the behavior: When I tested this, as long as the confirmation prompt was visible, there was no logging happening. It was only after I dismissed the confirmation prompt that it started logging the next events. Is that expected?

Reply:
Yeah, I don't think there's a good way for us to know when the prompt is _dismissed_. That was part of the motivation for doing the follow-up logging at intervals.

Reply:
Well it was more that nothing was logged while the prompt was visible. So as far as how it's described here:

>It will then log `User prompted before navigation and still on page` events at 5, 15, and 30 seconds after the onbeforeunload handler is called 

In reality it was more like:

>It will then log `User prompted before navigation and still on page` events at 5, 15, and 30 seconds after the onbeforeunload handler dialog is dismissed